js/main.js

```diff
@@ -1,2 +1,47 @@
 /* global data */
 /* exported data */
+var $entryForm = document.querySelector('#entry-form');
+var $photoUrl = document.querySelector('#photoURL');
+var $img = document.querySelector('.img');
+
+function photoInput(event) {
+  var url = $photoUrl.value;
+  $img.setAttribute('src', url);
+}
+
+$photoUrl.addEventListener('input', photoInput);
+
+function onSubmit(event) {
+  event.preventDefault();
+  var newEntry = {
+    entryTitle: $entryForm.elements.title.value,
+    entryURL: $entryForm.elements.photoURL.value,
+    entryNotes: $entryForm.elements.notes.value,
+    entryId: data.nextEntryId
+  };
+  data.nextEntryId++;
+  data.entries.unshift(newEntry);
+  $img.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $entryForm.reset();
+}
+
+$entryForm.addEventListener('submit', onSubmit);
+
+var $myStorage = window.localStorage;
+
+var previousDataObj = $myStorage.getItem('local-data-object');
+
+if (previousDataObj.entries !== null) {
+  var previousData = JSON.parse(previousDataObj);
+  data.entries = previousData.entries;
+  data.view = previousData.view;
+  data.editing = previousData.editing;
+  data.nextEntryId = previousData.nextEntryId;
```

Comment:
Instead of doing every property manually, you can just replace the entire value of `data`.